src/main/java/seedu/duke/commands/DeleteAccountCommand.java

```diff
@@ -37,21 +42,41 @@ private void processCommand() throws InvalidDeleteAccountCommandException {
 
     }
 
-    private void printSuccess(Ui ui) {
+    private void printSuccess (Ui ui) {
         ui.printf(Message.SUCCESSFUL_DELETE_ACCOUNT_COMMAND.getMessage(), this.currency.name());
     }
 
+    /**
+     * delete all related currency transaction in the main transaction list
+     *
+     * @param currency currency to be delete
+     */
+
+    private void deleteCurrencyTransaction (Currency currency) {
+        ArrayList<Transaction> transactionList = transactions.getTransactionList();
+        ArrayList<Transaction> transactionListToBeDeleted = new ArrayList<>();
+        for (Transaction transaction : transactionList) {
+            if (transaction.getCurrency() == currency) {
+                transactionListToBeDeleted.add(transaction);
+            }
+        }
+        for (Transaction transactionToBeDeleted : transactionListToBeDeleted) {
+            transactions.deleteTransaction(transactionToBeDeleted);
```

Comment:
There should be no need to iterate twice, since it works this way it should also work if you delete while iterating through the original transactionList